src/config/networkConfig.ts

```diff
@@ -342,17 +347,31 @@ export class NetworkConfig {
     return networks[this.networkId].earliestBlockToCheck
   }
 
-  getTokenFromAddress(address: string): Token {
+  getTokenFromAddress(address: string): Maybe<Token> {
     const tokens = networks[this.networkId].tokens
 
-    for (const token of tokens) {
-      const tokenAddress = token.address
-      if (tokenAddress.toLowerCase() === address.toLowerCase()) {
-        return token
+    const tokenDefault = tokens.find((token: Token) =>
+      token.address.toLowerCase().includes(address.toLowerCase())
+    )
```

Comment:
This is kind of duplicated, could be written as `searchBy(defaultTokens, 'address', address), || searchBy(uniswapTokens, 'address', address) || null` and for the other case the same but with searching by symbol.

The find result in something like `this.listFrom.find(token => token[key].toLowerCase().includes(value.toLowerCase()) && token.chainId === this.networkId` 

And it is possible to filter in `getTokensFromUniswap` by networkId to avoid the last condition that only has sense with passing the list of uniswap and not the default list.